server.py

```diff
@@ -115,15 +119,16 @@ def changeState(self, newstate: ChannelState):
             self.join_event.clear()
             self.leave_event.set()
 
-        self.onStateChange()
 
     def onStateChange(self):
-        if self._target_state == True and self._state == ChannelState.PARTED:
-            # first wait 0 seconds, but if that fails, wait longer
-            self._wait_join_task = asyncio.ensure_future(self.joinAfterDelay(self._wait_time))
-            self._wait_time = RETRY_WAIT
-        elif self._target_state == False and self._state == ChannelState.JOINED:
-            self.sendPart()
+        if not self._target_state:
```

Comment:
Looks like you misunderstood this if statement: we want to join the channel if target state is true, and we want to part the channel if target state is false.